src/metabase/util/date_2.clj

```diff
@@ -88,6 +96,25 @@
   ;; replace the `T` with a space. Easy!
   (str/replace-first (format t) #"(\d{2})T(\d{2})" "$1 $2"))
 
+(defn format-human-readable
+  "Format a temporal value `t` in a human-friendly way for `locale` (by default, the current User's locale).
+
+    (format-human-readable #t \"2021-04-02T14:42:09.524392-07:00[US/Pacific]\" \"es-MX\")
+    ;; -> \"abril 2 2:42 PM (Hora de verano del Pacífico)\""
+  ;; TODO -- need to figure out how to have this do 12-hour vs 24-hour time depending on Locale.
```

Comment:
It's more than 12/24-hour formatting. These long form formats differ a lot between locales, and it would look significantly more polished if we could handle them accurately.

That can be accomplished with date-time formatters created with `java.time.format.DateTimeFormatter/ofLocalized{Date,Time,DateTime}`, e.g.

```clojure
(let [formatter (-> (DateTimeFormatter/ofLocalizedDateTime FormatStyle/LONG)
                    (.withLocale Locale/GERMAN))]
  (t/format formatter (t/zoned-date-time)))
; => "3. April 2021 um 04:19:02 SAST"
```

There doesn't seem to be formatters for `OffsetTime` and `OffsetDateTime`, but we could perhaps create zoned objects from those.

Reply:
@walterl that makes a lot of sense. I'll change it